exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1147 @@
+func (d *Deribit) UpdateOrderbook(ctx context.Context, p currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	book := &orderbook.Base{
+		Exchange:        d.Name,
+		Pair:            p,
+		Asset:           assetType,
+		VerifyOrderbook: d.CanVerifyOrderbook,
+	}
+	fmtPair, err := d.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	var obData *Orderbook
+	if d.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+		obData, err = d.WSRetrieveOrderbookData(fmtPair.String(), 50)
+	} else {
+		obData, err = d.GetOrderbookData(ctx, fmtPair.String(), 50)
+	}
+	if err != nil {
+		return nil, err
+	}
+	book.Asks = make([]orderbook.Item, len(obData.Asks))
+	for x := range book.Asks {
+		book.Asks[x] = orderbook.Item{
+			Price:  obData.Asks[x][0],
+			Amount: obData.Asks[x][1],
+		}
+		if book.Asks[x].Price == 0 {
+			return nil, fmt.Errorf("%w, ask price=%f", errInvalidPrice, book.Asks[x].Price)
+		}
+	}
+	book.Bids = make([]orderbook.Item, len(obData.Bids))
+	for x := range book.Bids {
+		book.Bids[x] = orderbook.Item{
+			Price:  obData.Bids[x][0],
+			Amount: obData.Bids[x][1],
+		}
+		if book.Bids[x].Price == 0 {
+			return nil, fmt.Errorf("%w, bid price=%f", errInvalidPrice, book.Asks[x].Price)
+		}
+	}
+	err = book.Process()
+	if err != nil {
+		return book, err
+	}
+	return orderbook.Get(d.Name, p, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (d *Deribit) UpdateAccountInfo(ctx context.Context, _ asset.Item) (account.Holdings, error) {
+	var resp account.Holdings
+	resp.Exchange = d.Name
+	currencies, err := d.GetCurrencies(ctx)
+	if err != nil {
+		return resp, err
+	}
+	resp.Accounts = make([]account.SubAccount, len(currencies))
+	for x := range currencies {
+		var data *AccountSummaryData
+		if d.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+			data, err = d.WSRetrieveAccountSummary(currencies[x].Currency, false)
+		} else {
+			data, err = d.GetAccountSummary(ctx, currencies[x].Currency, false)
+		}
+		if err != nil {
+			return resp, err
+		}
+		var subAcc account.SubAccount
+		subAcc.Currencies = append(subAcc.Currencies, account.Balance{
+			Currency: currency.NewCode(currencies[x].Currency),
+			Total:    data.Balance,
+			Hold:     data.Balance - data.AvailableFunds,
+		})
+		resp.Accounts[x] = subAcc
+	}
+	return resp, nil
+}
+
+// FetchAccountInfo retrieves balances for all enabled currencies
+func (d *Deribit) FetchAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
+	creds, err := d.GetCredentials(ctx)
+	if err != nil {
+		return account.Holdings{}, err
+	}
+	accountData, err := account.GetHoldings(d.Name, creds, assetType)
+	if err != nil {
+		return d.UpdateAccountInfo(ctx, assetType)
+	}
+	return accountData, nil
+}
+
+// GetFundingHistory returns funding history, deposits and withdrawals
+func (d *Deribit) GetFundingHistory(ctx context.Context) ([]exchange.FundHistory, error) {
```

Comment:
`GetFundingRates` && `IsPerpetualFutureCurrency` can be implemented for futures. 

`https://docs.deribit.com/#public-get_funding_rate_value`